### PR TITLE
Service refactoring

### LINKED ISSUE #597: Patron service set up is broken
## Is your feature request related to a problem? Please describe

With the removal of the builder pattern, the setup of Patron has become a one-liner with the `New()` function.
Previously though, by using the builder, we created a builder first which set up everything and then we created the service and ran it.

The result is that logging and tracing are not set up correctly and the whole main function in the service is using the default logger.

## Is this a bug? Please provide steps to reproduce, a failing test etc.

Just by creating a service, the problem is apparent.

## Describe the solution

In order to avoid reintroducing the builder pattern again, we could keep the `New()` function which sets up a service (logging, tracing etc.) at the beginning of the main and provide the components to run in the `Run()` method. This would give us the flexibility of the functional option pattern without the extra builder struct. Another benefit is that we are now decoupling the HTTP component from the service. If the end user would like to set up a worker without an HTTP endpoint this was not possible because Patron by default created an HTTP service. Somebody might want to start without or even instead HTTP start a gRPC one.

### PR DESCRIPTION
**Pull request by mantzas**
## Which problem is this PR solving?

Closes #597.

## Short description of the changes

-

### CODE REVIEW

File: component/http/component.go [hunk 1]
```diff
@@ -109,11 +111,69 @@ func (c *Component) createHTTPServer() *http.Server {
...
 		WriteTimeout: c.httpWriteTimeout,
-		IdleTimeout:  httpIdleTimeout,
+		IdleTimeout:  defaultIdleTimeout,
 		Handler:      routerAfterMiddleware,
 	}
 }
 
+func port() (int, error) {
+	var err error
+	portVal := int64(defaultPort)
```
Comment (pkritiotis) [inline component/http/component.go]: Do we need an `int64` here?

Comment (pkritiotis) [reply component/http/component.go]: Also it seems that this line is redundant since this variable is then assigned to the parsed value of the env var

**Review by xenosdio — DISMISSED** [review]

File: component/http/v2/component.go [hunk 2]
```diff
@@ -6,21 +6,67 @@ import (
...
+	defaultReadTimeout         = 30 * time.Second
+	defaultWriteTimeout        = 60 * time.Second
+	defaultIdleTimeout         = 240 * time.Second
+	defaultHandlerTimeout      = 59 * time.Second // should be smaller than write timeout
+	defaultShutdownGracePeriod = 5 * time.Second
 )
 
+func port() (int, error) {
+	var err error
+	portVal := int64(defaultPort)
```
Comment (pkritiotis) [inline component/http/v2/component.go]: Same case here

**Review by pkritiotis — CHANGES_REQUESTED** [review]
Looks good, just a tiny comment

File: component/http/component.go (hunk 1 above)
Comment (mantzas) [reply component/http/component.go]: good catch!

File: component/http/v2/component.go (hunk 2 above)
Comment (mantzas) [reply component/http/v2/component.go]: good catch!

**Review by pkritiotis — APPROVED** [review]

**Review by xenosdio — APPROVED** [review]

**Pull request merged**